src/engine/sparqlExpressions/BlankNodeExpression.cpp

```diff
@@ -0,0 +1,167 @@
+//   Copyright 2025, University of Freiburg,
+//   Chair of Algorithms and Data Structures.
+//   Author: Robin Textor-Falconi <textorr@informatik.uni-freiburg.de>
+
+#include "engine/sparqlExpressions/BlankNodeExpression.h"
+
+#include <absl/strings/str_cat.h>
+
+#include "engine/sparqlExpressions/LiteralExpression.h"
+#include "engine/sparqlExpressions/NaryExpressionImpl.h"
+#include "engine/sparqlExpressions/SparqlExpressionValueGetters.h"
+#include "util/ChunkedForLoop.h"
+
+namespace sparqlExpression {
+
+namespace detail {
+// If the char is allowed inside the blank node label. A little bit stricter
+// than the specification for simplicity.
+bool isAllowedChar(char c) {
+  return (c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z') || c == '_' ||
+         (c >= '0' && c <= '9');
+}
+
+// Escape a string for use as a blank node label.
+std::string escapeStringForBlankNode(std::string_view input) {
+  std::string output;
+  for (char c : input) {
+    if (isAllowedChar(c)) {
+      output += c;
+    } else {
+      // Since we don't allow the '.' char in our `isAllowedChar` check, we can
+      // safely use it to denote escape sequences.
+      output += ".";
+      output += std::to_string(c);
+    }
```

Comment:
1. 
Add a `reserve(input.size()` (or count the not allowed characters in advance).
2. Use `absl::StrAppend`
3. I will think of something else:)